packages/core-entities/src/models/views/YieldViews.ts

```diff
@@ -430,6 +444,188 @@ export const YieldViews = (self: Instance<typeof NetworkModel>) => {
+            liquidity: getLiquidity(t),
+            underlying: t.underlying
+              ? getTokenByID(t.underlying)
+              : undefined,
+            collateralFactor: getDebtOrCollateralFactor(t, false),
+          };
+        });
+    });
+    const tokenList = getUniqueUnderlyingSymbols(productGroupData);
+    const defaultSymbol = 'ETH';
+    return {
+      tokenList,
+      productGroupData: productGroupData,
+      defaultSymbol,
+    };
+  };
+
+  const getPortfolioStateZeroEarnData = () => {
+    const group = ['fCash', 'PrimeCash', 'nToken'];
+    const productGroupData = group.map((group) => {
+      return getTokensByType(group)
+        .filter((data) => !data?.isFCashDebt)
+        .map((t) => {
+          return {
+            token: t,
+            apy: getSpotAPY(t.id),
+            tvl: getTVL(t),
+            liquidity: getLiquidity(t),
+            underlying: t.underlying
+              ? getTokenByID(t.underlying)
+              : undefined,
+            collateralFactor: getDebtOrCollateralFactor(t, false),
+          };
+        });
+    });
+    const tokenList = getUniqueUnderlyingSymbols(productGroupData);
+    const highestAPYSymbol = getDefaultHighestAPYSymbol(productGroupData);
+
+    return {
+      tokenList,
+      productGroupData,
+      defaultSymbol: highestAPYSymbol,
+    };
+  };
+
+  const getLeveragedNTokenData = () => {
+    const leveragedNTokenData = getTokensByType('nToken').map((t) => {
+          const debtTokens = getDefaultLeveragedNTokenAPYs(t);
+          const leveragedNTokenData = debtTokens.reduce((max, current) => {
+            return current?.apy?.totalAPY &&
+              max?.apy?.totalAPY &&
+              current.apy.totalAPY > max.apy.totalAPY
+              ? current
+              : max;
+          }, debtTokens[0]);
+        return {
+          token: t,
+          apy: leveragedNTokenData?.apy,
+          tvl: getTVL(t),
+          maxLeverageRatio: getLeverageRatios(t).maxLeverageRatio,
+          liquidity: getLiquidity(t),
+          underlying: t.underlying ? getTokenByID(t.underlying) : undefined,
+          collateralFactor: getDebtOrCollateralFactor(t, false),
+          debtToken: leveragedNTokenData?.debtToken
+        };
+      });
+      return leveragedNTokenData
+  }
+
+  const getAllListedVaultsData = () => {
+    const listedVaults = getAllListedVaults(true)
+    const listedVaultAddresses = listedVaults.map((v) => v.vaultAddress)
+
+    const allListedVaultsData = getTokensByType('VaultShare')
+    .filter((t) => t.vaultAddress && listedVaultAddresses.includes(t.vaultAddress))
+    .map((t) => {
+          const data = getDefaultVaultAPYs(t.vaultAddress || '');
+          const vaultAPYs = data.reduce((max, current) => {
+            return (current.apy.totalAPY || 0) > (max.apy.totalAPY || 0)
+              ? current
+              : max;
+          }, data[0])?.apy;
+          const vaultDebtToken = data.reduce((max, current) => {
```

Comment:
Why do the reduce twice? Isn't the correct debtToken already sitting in the vaultAPYs data structure?